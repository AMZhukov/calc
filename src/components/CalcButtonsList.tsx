import React from 'react';

type CalcButtonsListProps = {
    buttonsContent: string[],
    // eslint-disable-next-line no-unused-vars
    onClickButton: (event: React.SyntheticEvent<HTMLButtonElement>) => void
}

export const CalcButtonsList: React.FC<CalcButtonsListProps> = ({ buttonsContent, onClickButton }) => {
  return (
    <ul className="calc__action-buttons-list" data-testid="calculatorButtons">
      {buttonsContent.map((item) => {
        return (
          <li className="calc__action-button-wrapper" key={item}>
            <button
              className="calc__action-button"
              value={item}
              onClick={onClickButton}
              type="button"
            >
              {item}
            </button>
          </li>
        );
      })}
    </ul>
  );
};
