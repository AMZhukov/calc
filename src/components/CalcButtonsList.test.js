import React from 'react';
import '@testing-library/jest-dom/extend-expect'
import { render, screen } from '@testing-library/react';
import { CalcButtonsList } from "./CalcButtonsList";
import { BUTTONS_CONTENT } from "../constants/buttonsContent";

test('render Calculator buttons', () => {
    render(<CalcButtonsList buttonsContent={BUTTONS_CONTENT} onClickButton={() => {}}/>);

    const calculatorButtons = screen.getByTestId('calculatorButtons');
    expect(calculatorButtons).toBeInTheDocument();
    expect(calculatorButtons).toMatchSnapshot();
})
