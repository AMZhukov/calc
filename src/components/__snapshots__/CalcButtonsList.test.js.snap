// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Calculator buttons 1`] = `
<ul
  class="calc__action-buttons-list"
  data-testid="calculatorButtons"
>
  <li
    class="calc__action-button-wrapper"
  >
    <button
      class="calc__action-button"
      type="button"
      value="C"
    >
      C
    </button>
  </li>
  <li
    class="calc__action-button-wrapper"
  >
    <button
      class="calc__action-button"
      type="button"
      value="√"
    >
      √
    </button>
  </li>
  <li
    class="calc__action-button-wrapper"
  >
    <button
      class="calc__action-button"
      type="button"
      value="%"
    >
      %
    </button>
  </li>
  <li
    class="calc__action-button-wrapper"
  >
    <button
      class="calc__action-button"
      type="button"
      value="/"
    >
      /
    </button>
  </li>
  <li
    class="calc__action-button-wrapper"
  >
    <button
      class="calc__action-button"
      type="button"
      value="7"
    >
      7
    </button>
  </li>
  <li
    class="calc__action-button-wrapper"
  >
    <button
      class="calc__action-button"
      type="button"
      value="8"
    >
      8
    </button>
  </li>
  <li
    class="calc__action-button-wrapper"
  >
    <button
      class="calc__action-button"
      type="button"
      value="9"
    >
      9
    </button>
  </li>
  <li
    class="calc__action-button-wrapper"
  >
    <button
      class="calc__action-button"
      type="button"
      value="×"
    >
      ×
    </button>
  </li>
  <li
    class="calc__action-button-wrapper"
  >
    <button
      class="calc__action-button"
      type="button"
      value="4"
    >
      4
    </button>
  </li>
  <li
    class="calc__action-button-wrapper"
  >
    <button
      class="calc__action-button"
      type="button"
      value="5"
    >
      5
    </button>
  </li>
  <li
    class="calc__action-button-wrapper"
  >
    <button
      class="calc__action-button"
      type="button"
      value="6"
    >
      6
    </button>
  </li>
  <li
    class="calc__action-button-wrapper"
  >
    <button
      class="calc__action-button"
      type="button"
      value="-"
    >
      -
    </button>
  </li>
  <li
    class="calc__action-button-wrapper"
  >
    <button
      class="calc__action-button"
      type="button"
      value="1"
    >
      1
    </button>
  </li>
  <li
    class="calc__action-button-wrapper"
  >
    <button
      class="calc__action-button"
      type="button"
      value="2"
    >
      2
    </button>
  </li>
  <li
    class="calc__action-button-wrapper"
  >
    <button
      class="calc__action-button"
      type="button"
      value="3"
    >
      3
    </button>
  </li>
  <li
    class="calc__action-button-wrapper"
  >
    <button
      class="calc__action-button"
      type="button"
      value="+"
    >
      +
    </button>
  </li>
  <li
    class="calc__action-button-wrapper"
  >
    <button
      class="calc__action-button"
      type="button"
      value="00"
    >
      00
    </button>
  </li>
  <li
    class="calc__action-button-wrapper"
  >
    <button
      class="calc__action-button"
      type="button"
      value="0"
    >
      0
    </button>
  </li>
  <li
    class="calc__action-button-wrapper"
  >
    <button
      class="calc__action-button"
      type="button"
      value=","
    >
      ,
    </button>
  </li>
  <li
    class="calc__action-button-wrapper"
  >
    <button
      class="calc__action-button"
      type="button"
      value="="
    >
      =
    </button>
  </li>
</ul>
`;
